From 7e5dd8fdb7fec801bbd31cfadac4b099268f5641 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Fri, 27 Apr 2018 15:32:51 -0700
Subject: [PATCH 2/2] ni: zynq: Mender.io support for NI Ettus Research E31x
 series SDRs

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

This reverts commit ec45c2e3a2bd50a858e7bb97429bdd492cd2057e.
---
 configs/ni_e31x_sg1_defconfig | 2 +-
 configs/ni_e31x_sg3_defconfig | 2 +-
 include/configs/ni_e31x.h     | 9 +++------
 3 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/configs/ni_e31x_sg1_defconfig b/configs/ni_e31x_sg1_defconfig
index 6b91093f5f..1849e9b2be 100644
--- a/configs/ni_e31x_sg1_defconfig
+++ b/configs/ni_e31x_sg1_defconfig
@@ -8,7 +8,7 @@ CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-e31x-sg1"
 CONFIG_DEBUG_UART=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
-CONFIG_SYS_EXTRA_OPTIONS="ENV_IS_IN_MMC"
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV,ENV_IS_IN_MMC"
 CONFIG_BOOTDELAY=0
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/configs/ni_e31x_sg3_defconfig b/configs/ni_e31x_sg3_defconfig
index 627f1a3e29..ae5099ebff 100644
--- a/configs/ni_e31x_sg3_defconfig
+++ b/configs/ni_e31x_sg3_defconfig
@@ -8,7 +8,7 @@ CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-e31x-sg3"
 CONFIG_DEBUG_UART=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
-CONFIG_SYS_EXTRA_OPTIONS="ENV_IS_IN_MMC"
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV,ENV_IS_IN_MMC"
 CONFIG_BOOTDELAY=0
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_e31x.h b/include/configs/ni_e31x.h
index 7cd102cc81..d63b5c5f09 100644
--- a/include/configs/ni_e31x.h
+++ b/include/configs/ni_e31x.h
@@ -12,20 +12,17 @@
 
 #include <configs/zynq-common.h>
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE           CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET              0xE0000
-
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
 	"fit_image=boot/fitImage\0"	\
 	"mboard=e31x-sg3\0"	\
-	"bootargs=root=/dev/mmcblk0p2 rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
+	"bootargs=root=${mender_kernel_root} rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
 	"initrd_high=0x20000000\0"      \
-	"sdboot=echo Copying FIT from SD to RAM... && " \
+	"sdboot=run mender_setup; " \
+		"echo Copying FIT from SD to RAM... && " \
 		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
-- 
2.17.0

